django/db/models/fields/__init__.py

```diff
@@ -828,18 +828,17 @@ def get_flatchoices(self, include_blank=True,
         first_choice = blank_choice if include_blank else []
         return first_choice + list(self.flatchoices)
 
+    @warn_about_renamed_method('Field', '_get_val_from_obj', 'value_from_object',
+                               RemovedInDjango21Warning)
     def _get_val_from_obj(self, obj):
-        if obj is not None:
```

Comment:
I agree with @timgraham - there's simply no value in changing the implementation of a method that is being deprecated, even if the change in implementation seems unlikely to cause a problem in real code. (You might be surprised what "unlikely to break anything" changes we've seen break things.) The point of a deprecation path is to leave the existing API completely unchanged while pointing people to a new and better API, so we can (in a couple releases) remove the old one.

This is clearly not a case of "can't make any changes to anything because of back-compat" - we're deprecating a method, that's a change!